lib/gitsh/interpreter.rb

```diff
@@ -3,21 +3,31 @@
 
 module Gitsh
   class Interpreter
-    def initialize(env, options={})
-      @env = env
+    def initialize(options)
+      @env = options.fetch(:env)
       @parser_factory = options.fetch(:parser_factory, Parser)
+      @input_strategy = options.fetch(:input_strategy)
     end
 
+    def run
+      input_strategy.setup
```

Comment:
Should we check a return value here?

Update: I see, we're doing that in `Interpreter#run`.